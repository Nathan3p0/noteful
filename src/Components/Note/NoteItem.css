li.noteItem {
    background-color: white;
    height: 90px;
    border: 1px solid black;
    margin-bottom: 10px;
    padding: 10px;
    position: relative;
    max-width: 380px;
}


li.noteItem h2 {
    margin: 0;
    max-width: 80%;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

li.noteItem a {
    text-decoration: none;
}

li.noteItem .noteDate {
    position: absolute;
    bottom: 10px;
}

li.noteItem .btnDelete {
    position: absolute;
    right: 31px;
    top: 0px;
    transform-origin: top right;
    transform: rotate(-90deg);
    height: 30px;
    width: 88px;
}