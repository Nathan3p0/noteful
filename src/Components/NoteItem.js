import React from 'react';

const NoteItem = (props) => {
    return ( 
        <li>Notes
        Date Modified
        <button>Delete Note</button>
        </li>
     );
}
 
export default NoteItem;