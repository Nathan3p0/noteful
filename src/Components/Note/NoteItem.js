import React from 'react';
import { Link } from 'react-router-dom';
import DeleteNoteButton from '../DeleteNoteButton/DeleteNoteButton';
import PropTypes from 'prop-types';
import './NoteItem.css'

const NoteItem = (props) => {
    
    return ( 
                <li className="noteItem"><Link to={`/note/${props.id}`}><h2>{props.name}</h2></Link>
                <span className="noteDate">Date Modified on {new Date(props.modified).toLocaleDateString('en-US', { day: 'numeric', month: 'short', year: 'numeric'  })}</span>
                <DeleteNoteButton noteId={props.id} />
            </li>
     );
}

NoteItem.propTypes = {
    id : PropTypes.string.isRequired,
    name : PropTypes.string.isRequired,
    modified : PropTypes.string
}
 
export default NoteItem;