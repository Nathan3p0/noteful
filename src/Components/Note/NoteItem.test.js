import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom';
import NoteItem from './NoteItem'


describe('Note Item Component', () => {

    const noteProps = {
        id: "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        name: "Dogs",
        modified: "2019-01-03T00:00:00.000Z",
    }

    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter>
            <NoteItem />
        </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders empty given no note info', () => {
        const tree = renderer.create(
        <BrowserRouter>
            <NoteItem />
        </BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot()
    })
    
    it('renders properly when given props', () => {
        const tree = renderer.create(
        <BrowserRouter>
            <NoteItem id={noteProps.id} name={noteProps.name} modified={noteProps.modified} />
        </BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot()
    })

})