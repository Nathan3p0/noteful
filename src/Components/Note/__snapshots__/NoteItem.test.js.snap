// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Item Component renders empty given no note info 1`] = `
<li
  className="noteItem"
>
  <h2>
    <a
      href="/note/"
      onClick={[Function]}
    >
      
    </a>
  </h2>
  <span
    className="noteDate"
  >
    Date Modified on 
    Invalid Date
  </span>
  <button
    className="btn btnDelete"
    onClick={[Function]}
  >
    Delete Note
  </button>
</li>
`;

exports[`Note Item Component renders properly when given props 1`] = `
<li
  className="noteItem"
>
  <h2>
    <a
      href="/note/cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1"
      onClick={[Function]}
    >
      Dogs
    </a>
  </h2>
  <span
    className="noteDate"
  >
    Date Modified on 
    Jan 2, 2019
  </span>
  <button
    className="btn btnDelete"
    onClick={[Function]}
  >
    Delete Note
  </button>
</li>
`;
